webknossos-tracingstore/app/com/scalableminds/webknossos/tracingstore/tracings/volume/VolumeDataZipHelper.scala

```diff
@@ -0,0 +1,73 @@
+package com.scalableminds.webknossos.tracingstore.tracings.volume
+
+import java.io.{File, FileOutputStream, InputStream}
+
+import com.scalableminds.util.geometry.Point3D
+import com.scalableminds.util.io.ZipIO
+import com.scalableminds.util.tools.ByteUtils
+import com.scalableminds.webknossos.datastore.dataformats.wkw.WKWDataFormatHelper
+import com.scalableminds.webknossos.datastore.models.BucketPosition
+import com.scalableminds.webknossos.wrap.WKWFile
+import net.liftweb.common.Box
+import org.apache.commons.io.IOUtils
+
+import scala.collection.mutable
+import scala.concurrent.ExecutionContext
+
+trait VolumeDataZipHelper extends WKWDataFormatHelper with ByteUtils {
+
+  protected def withBucketsFromZip(zipFile: File)(block: (BucketPosition, Array[Byte]) => Unit)(
+      implicit ec: ExecutionContext): Box[Unit] = {
+    val unzipResult = ZipIO.withUnziped(zipFile) {
+      case (fileName, is) =>
+        WKWFile.read(is) {
+          case (header, buckets) =>
+            if (header.numBlocksPerCube == 1) {
+              parseWKWFilePath(fileName.toString).map { bucketPosition: BucketPosition =>
+                if (buckets.hasNext) {
+                  val data = buckets.next()
+                  if (!isAllZero(data)) {
+                    block(bucketPosition, data)
+                  }
+                }
+              }
+            }
+        }
+    }
+    for {
+      _ <- unzipResult
+    } yield ()
```

Comment:
I find it hard to understand what this does. Is it the same as `unzipResult.map(_ => ())`? Why do we want to change the return type? Is it meant to deter someone from using the return value?

Reply:
`map` is fine with me :) It does not make sense to use the return value, so I wanted to hide it in the signature, yes.